app/services/concerns/course/user_invitation_service/email_invitation_concern.rb

```diff
@@ -7,17 +7,23 @@ module Course::UserInvitationService::EmailInvitationConcern
 
   # Invites users to the given course.
   #
+  # The result of the transaction is both saving the course as well as validating validity
+  # because Rails does not handle duplicate nested attribute uniqueness constraints.
+  #
   # @param [Array<Hash>|File|TempFile] users Invites the given users.
   # @return [Boolean] True if the invitations were successfully created and sent out. The errors
   #   that this method would add to the provided course is in the +course_users+ association.
   # @raise [CSV::MalformedCSVError] When the file provided is invalid.
   def invite(users)
-    Course.transaction do
-      registered_users, invited_users = invite_from_source(users)
+    registered_users = nil
+    invited_users = nil
 
-      break false if !@current_course.save || !@current_course.valid?
-      send_invitation_emails(registered_users, invited_users)
+    success = Course.transaction do
```

Comment:
See the comment above... It works only when the job is synchronous. Otherwise it will give a record not found because Sidekiq ru

In app/services/concerns/course/user_invitation_service/email_invitation_concern.rb<https://github.com/Coursemology/coursemology2/pull/1028#discussion_r60864845>:

>
> -      break false if !@current_course.save || !@current_course.valid?
> -      send_invitation_emails(registered_users, invited_users)
> +    success = Course.transaction do


@lowjoel<https://github.com/lowjoel> Anything wrong with the old implementation ?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Coursemology/coursemology2/pull/1028/files/9aec0fe7c2831ffdcc97266d0653ab5e80aafba6#r60864845>